posthog/hogql_queries/insights/trends/aggregation_operations.py

```diff
@@ -138,7 +138,18 @@ def _math_func(self, method: str, override_chain: Optional[list[str | int]]) ->
         else:
             chain = ["properties", self.series.math_property]
 
-        return ast.Call(name=method, args=[ast.Field(chain=chain)])
+        return ast.Call(
+            # Two caveats here:
+            # 1. We always parse/convert the value to a Float64, to make sure it's a number. This truncates precision
+            # of very large integers, but it's a tradeoff preventing queries failing with "Illegal type String"
+            # 2. We fall back to 0 when there's no data, which is not quite kosher for math functions other than sum
```

Comment:
This is a good point - haven't looked into this at all but would there by any problem using something like `maxIf(prop, (x) -> notNull(x))?` instead of max in this case?

Reply:
That's how this works already actually, `max()` omits null values. The trouble is only when showing the results to a user, when there's no matching events e.g. `min()` should be `null` ideally (but how to chart that crap?) or perhaps `infinity` (also confusing and hard to visualize)